src/d/actor/d_a_player_particle.inc

```diff
@@ -106,22 +136,147 @@ void daPy_swimTailEcallBack_c::draw(JPABaseEmitter*) {
+    GXSetVtxDesc(GX_VA_TEX0, GX_DIRECT);
+    GXSetVtxAttrFmt(GX_VTXFMT0, GX_VA_POS, GX_CLR_RGBA, GX_S8, 0);
+    GXSetVtxAttrFmt(GX_VTXFMT0, GX_VA_TEX0, GX_CLR_RGBA, GX_U8, 0);
+
+    GXTexObj texObj;
+    GXColor color;
+    if (mLockFlag) {
+        GXInitTexObj(&texObj, mpLockTex, mpImg->width, mpImg->height, (GXTexFmt)mpImg->format, (GXTexWrapMode)mpImg->wrapS, (GXTexWrapMode)mpImg->wrapT, (mpImg->mipmapCount > 1) ? GX_TRUE : GX_FALSE);
+
+        color.r = 0xFF;
+        color.g = 0xFF;
+        color.b = 0x32;
+        color.a = mLockAlpha;
+        GXSetTevColor(GX_TEVREG0, color);
+        GXSetTevColor(GX_TEVREG1, color);
+    } else {
+        GXInitTexObj(&texObj, mpSightTex, 0x80, 0x80, GX_TF_I4, GX_CLAMP, GX_CLAMP, GX_FALSE);
+
+        color.r = 0xFF;
+        color.g = 0x00;
+        color.b = 0x00;
+        color.a = 0xFF;
+        GXSetTevColor(GX_TEVREG0, color);
+        GXSetTevColor(GX_TEVREG1, color);
+    }
+
+    GXInitTexObjLOD(&texObj, GX_LINEAR, GX_LINEAR, 0.0f, 0.0f, 0.0f, GX_FALSE, GX_FALSE, GX_ANISO_1);
+    GXLoadTexObj(&texObj, GX_TEXMAP0);
+    GXLoadPosMtxImm(mMtx, GX_PNMTX0);
+    GXSetCurrentMtx(GX_PNMTX0);
+    GXSetTevOrder(GX_TEVSTAGE0, GX_TEXCOORD0, GX_TEXMAP0, GX_COLOR_NULL);
+    GXCallDisplayList(&l_sightDL, sizeof(l_sightDL) - 0x09);
+    J3DShape::resetVcdVatCache();
 }
 
 /* 8012952C-801296FC       .text setSight__18daPy_sightPacket_cFv */
 void daPy_sightPacket_c::setSight() {
-    /* Nonmatching */
+    cXyz proj;
+    mDoLib_project(&mPos, &proj);
+    mDoMtx_stack_c::transS(proj);
+
+    if (mLockFlag) {
+        f32 var_f2;
+        f32 var_f31;
+        if (field_0x6 < 13) {
+            var_f2 = 1.0f - ((s16)field_0x6 / 13.0f);
+            var_f31 = -var_f2;
+        } else {
+            var_f2 = (field_0x6 - 13) / 13.0f;
+            var_f31 = var_f2;
+        }
+
+        mLockAlpha = 150.0f + (30.0f * var_f2);
+        f32 temp_f3 = 0.65f + (0.35f * var_f2);
+
+        mDoMtx_stack_c::scaleM(temp_f3 * mpImg->width, temp_f3 * mpImg->height, temp_f3);
+        mDoMtx_stack_c::ZrotM(32768.0f * cM_ssin(16384.0f * var_f31));
+    } else {
+        mDoMtx_stack_c::scaleM(10.0f, 10.0f, 10.0f);
+    }
+
+    mDoMtx_copy(mDoMtx_stack_c::get(), mMtx);
+    dComIfGd_set2DXlu(this);
 }
 
 /* 801296FC-801297F4       .text initSwBlur__13daPy_swBlur_cFPA4_fifi */
-void daPy_swBlur_c::initSwBlur(MtxP, int, f32, int) {
-    /* Nonmatching */
+void daPy_swBlur_c::initSwBlur(MtxP mtx, int idx, f32 blurTopRate, int blurColorType) {
+    int i = idx * 2;
+
+    mBlurTopRate = blurTopRate;
+    mBlurColorType = blurColorType;
+
+    mDoMtx_multVec(mtx, mpPosBuffer + i, &field_0x034[0]);
+    mDoMtx_multVec(mtx, mpPosBuffer + i + 1, &field_0x304[0]);
+
+    field_0x034[0] += (field_0x034[0] - field_0x304[0]) * mBlurTopRate;
+    field_0x034[1] = field_0x034[0];
+    field_0x304[1] = field_0x304[0];
+    field_0x014 = 0;
+    field_0x018 = idx;
+    field_0x028.set(mtx[0][3], mtx[1][3], mtx[2][3]);
```

Comment:
this might be
```suggestion
    mDoMtx_multVecZero(mtx, &field_0x028);
```

Reply:
Didn't realize that function existed, but it doesn't seem to work here.

Reply:
that's slightly surprising, it works in setDamageFlameEmitter in this file so I assumed they would have used it consistently. does it work for the temp cXyzs?

Reply:
It seems to work for most of them.